### PR TITLE
CSAM IWF reporting from Aselo: API integration

### PR DESCRIPTION
**Pull request by GPaoloni**
Jira: https://bugs.benetech.org/browse/CHI-704

This PR is related to:
- https://github.com/techmatters/flex-plugins/pull/543
- https://github.com/techmatters/hrm/pull/131

This PR adds a new function (`reportToIWF`) to submit reports to IWF from our system.
It expects the following inputs:
- `Reported_URL` **required**: a string representing the url to the content being reported.
- `Reporter_Anonymous` **required**: either Y or N to indicate if the report is anonymous or not
- `Reporter_First_Name`: reporter first name (used if not anonymous).
- `Reporter_Last_Name` reporter last name (used if not anonymous).
- `Reporter_Email_ID`: reporter email address (**required** if not anonymous).
- `Reporter_Description` report brief description.

~WIP:~
- ~Unit tests.~
- ~Move the function to be protected before merging (so it's not public).~
- ~Include IWF credentials on deploy workflow.~

To test:
- Checkout this branch.
- Create a `.env` file in the root folder and fill this variables 
  - IWF_API_USERNAME
  - IWF_API_PASSWORD
  - IWF_API_URL
 
  (look for the values on Parameter Store using the same names).
- Run `npm ci` to install deps and then `npm start` to start local server.
- Hit http://localhost:3000/reportToIWF?Reported_URL=https://something.com&Reporter_Anonymous=Y in your browser or using curl
  ```
  curl -X POST -H 'Content-type: application/json' --data '{"Reported_URL": "something", "Reporter_Anonymous": "Y"}' http://localhost:3000/reportToIWF
  ```
- Play around using different values for the parameters.

What to expect: the function only rejects the request prior to hit IWF API if the **required** parameters are missing or invalid, but sends it otherwise, returning the same response that IWF sends. This means that, if for example, we submit a non-anonymous report without `Reporter_Email_ID`, the function will hit IWF but this will be rejected. We can validate this for ourselves but IWF is already doing some work on the inputs, like checking that email is a valid address. If we go with this approach, we rely on IWF validations, saving ourselves from this work.
Example response when sending a non anonymous report with a non empty but invalid email address (asdasd@asdasd).
Screenshot from 2021-11-10 14-24-38
Each input has it's own limitations and IWF performs the pertinent checks, so I don't think we should duplicate all of those, but let me know if you think different!

### CODE REVIEW

**Review by murilovmachado — APPROVED** [review]
Code looks good to me!

File: functions/reportToIWF.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+  Submission_Type_ID: 1; // 1 for online report
+  Reported_Category_ID: 2; // 2 for suspected child sexual abuse report (remit child)
+  Reported_URL: string; // Max 1000 characters
+  Reporter_Anonymous: 'Y' | 'N'; // Is the report anonymous or not
+  Reporter_First_Name: string | null; // Max 50 characters
+  Reporter_Last_Name: string | null; // Max 50 characters
+  Reporter_Email_ID: string | null; // Max 100 characters
+  Reporter_Country_ID: number | null; // Reporter's country (Helpline specific)
+  Reporter_Description: string | null; // Max 500 characters
+};
```
[the remark this answers is not in the document]
Comment (GPaoloni) [reply functions/reportToIWF.ts:37]: Yes, they send back an error within the json response, like
`"responseDescription":"Error: Reporter First Name - maximum 50 characters allowed exceeded"`

**Comment by GPaoloni** [comment]
@nickhurlburt @murilovmachado [In this commit](https://github.com/techmatters/serverless/pull/142/commits/47631eddf254d5340404f741053d4afe2dc24304) I'm introducing IWF credentials as part of the deployment action, which I'm not sure is correct as not every helpline will be using them. Do you think this is fine for now or is it better to address now?
Probably related to https://github.com/techmatters/serverless/pull/143#discussion_r756206173

**Comment by nickhurlburt** [comment]
@GPaoloni looking back at the comments on https://github.com/techmatters/serverless/pull/143 about naming of `SAFERNET_ENDPOINT`, I don't think we have a clear idea of this. A good topic for discussion.

**Comment by murilovmachado** [comment]
> @nickhurlburt @murilovmachado [In this commit](https://github.com/techmatters/serverless/pull/142/commits/47631eddf254d5340404f741053d4afe2dc24304) I'm introducing IWF credentials as part of the deployment action, which I'm not sure is correct as not every helpline will be using them. Do you think this is fine for now or is it better to address now? 

@GPaoloni I think we can merge like this just for the sake of getting this feature done, but I would suggest that the next immediate task (for me or you) would be to work on the custom action (as we discussed on Friday). It feels to me we can't postpone that.

**Pull request merged**